<!-- -->
<launch>

<arg name="laser_odom_topic_name" default="/lidar_eskf/odom"/>

<arg name="initial_x"       default="0.0"/>
<arg name="initial_y"       default="0.0"/>
<arg name="initial_z"       default="0.0"/>
<arg name="initial_yaw"     default="0.0"/>


<remap from="/fcs/odometry" to="/fcs/pose_estimation"/>


<node name="fcs"
      pkg="mav_fcs" type="mav_fcs" output="screen">
      
  <param name="initial_x" value="$(arg initial_x)"/>
  <param name="initial_y" value="$(arg initial_y)"/>
  <param name="initial_z" value="$(arg initial_z)"/>
  <param name="initial_yaw" value="$(arg initial_yaw)"/>
  
  
  <!-- path tracking control parameters -->
  
        <param name="laser_odom_topic_name" value="$(arg laser_odom_topic_name)"/>
  
        <param name="control_freq"      value="50.0" /> # make sure this is the odom freq
        <param name="DEBUG"             value="false" />
        <param name="USE_DJI_VEL"       value="false" /> 
        <param name="MAX_ROLL"          value="5.0" />  # in degrees
        <param name="MAX_PITCH"         value="5.0" />  # in degrees
        <param name="MAX_YAW_RATE"      value="10.0" />  # in degree/s
        <param name="MAX_Z_RATE"        value="0.3" />   # in meter/s
        
        <param name="MIN_ROLL"          value="0.0" />  # in degrees
        <param name="MIN_PITCH"         value="0.0" />  # in degrees
        <param name="MIN_YAW_RATE"      value="0.0" />  # in degree/s
        <param name="MIN_Z_RATE"        value="0.0" />  # in meter/s

        <param name="MAX_VEL_X"         value="0.5" />
        <param name="MAX_VEL_Y"         value="0.5" />
        <param name="MAX_VEL_Z"         value="0.3" />  # duplicated with MAX_Z_RATE, take the smaller

        <param name="MAX_VEL_INTG"      value="2.0" />
        <param name="MAX_POS_INTG"      value="0.2" />
        
        <param name="FW_ROLL"          value="0.0" />  # in degrees
        <param name="FW_PITCH"         value="0.0" />  # in degrees
        <param name="FW_Z_RATE"        value="0.0" />  # in meter/s
        
        <!-- VELOCITY CONTROL LOOP-->
        <param name="P_VEL_Y"           value="6.0" />
        <param name="P_VEL_X"           value="6.0" />
        
        <param name="I_VEL_Y"           value="0.0" />  
        <param name="I_VEL_X"           value="0.0" />  
        <param name="INIT_VEL_INTG_Y"   value="0.8" />  
        <param name="INIT_VEL_INTG_X"   value="0.0" />  

        <param name="D_VEL_Y"           value="0.0" />
        <param name="D_VEL_X"           value="0.0" />
    
        <!-- POSITION CONTROL LOOP-->
        <param name="P_POS_X"           value="0.4" /> #0.6
        <param name="P_POS_Y"           value="0.4" /> #0.6
        <param name="P_POS_Z"           value="0.8" /> #1.0
        
        <param name="I_POS_X"           value="0.0" /> #0.01
        <param name="I_POS_Y"           value="0.0" /> #0.01
        <param name="I_POS_Z"           value="0.0" /> #0.01
        
        <param name="D_POS_X"           value="0.0" /> #0.3
        <param name="D_POS_Y"           value="0.0" /> #0.3
        <param name="D_POS_Z"           value="0.0" /> #0.3
        
        <!-- ORIENTATION CONTROL LOOP-->
        <param name="P_ORI_Z"           value="1.0" /> 
        
        <!-- LASER FRAME AND DJI BODY FRAME YAW OFFSET-->
        <param name="YAW_OFFSET"        value="0.0" />
  
</node>

</launch>

